pyhyp/utils.py

```diff
@@ -38,15 +40,39 @@ def simpleOCart(nearFile, dh, hExtra, nExtra, sym, mgcycle, outFile, userOptions
         Output file name.
 
     userOptions : dict, optional
-        The name of the nearfield CGNS file to mesh around.
+        Custom pyhyp options to be used with this extrusion.
+
+    xBounds : array (2 x 3)
+        optional bounding box coordinates desired for the center cartesian grid.
+        It can be obtained by:
+            xMin, xMax = grid.getBoundingBox()
+        and then the xBounds array can be set as
+            xBounds = [xMin, xMax]
+        this is useful for modifying the xmin and xmax values rather than just using
+        the bounding box of the grid.
+
+    useFarfield : bool
+        optional flag to control the outermost layer's BC. Default, True, will result
+        in a farfield outer layer, setting this to false does overset BCs
+        on the outermost layer
 
     """
 
-    # Read the nearfield file
-    grid = readGrid(nearFile)
-
-    # First run simpleCart with no extension
-    X, dx = grid.simpleCart(dh, 0.0, 0, sym, mgcycle, outFile=None)
+    # check if we have a grid object as input or the filename
+    if type(inputGrid) == Grid:
+        # use the grid object as is
+        pass
```

Comment:
If we are not doing anything for this type, we don't need the check. Alternatively, we can add an else statement that raises `TypeError`.